src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Intermediate/ComponentAttributeIntermediateNode.cs

```diff
@@ -242,4 +243,69 @@ internal static bool TryGetEventCallbackArgument(ReadOnlyMemory<char> candidate,
         argument = default;
         return false;
     }
+
+    internal static bool TryGetActionArgument(ReadOnlyMemory<char> candidate, out ReadOnlyMemory<char> argument)
+    {
+        // Strip 'global::' from the candidate.
+        if (candidate.Span.StartsWith("global::".AsSpan()))
+        {
+            candidate = candidate["global::".Length..];
+        }
+
+        if (!candidate.Span.StartsWith("System.Action".AsSpan()))
+        {
+            argument = default;
+            return false;
+        }
+
+
+        candidate = candidate["System.Action".Length..];
+        if (candidate.Span is ['<', .., '>'])
+        {
+            argument = candidate[1..^1];
+            return true;
+        }
+
+        argument = default;
+        return false;
+    }
+
+    internal static bool TryGetGenericActionArgument(ReadOnlyMemory<char> candidate, string genericType, [NotNullWhen(true)] out string argument)
```

Comment:
Consider adding a doc comment with an example to this method, because it's not obvious what it's doing.